### PR TITLE
Vue analytic

### PR DESCRIPTION
**Pull request by GoodDayForSurf**

### CODE REVIEW

File: packages/vue/src/pages/analytics/components/analytics-toolbar.vue [hunk 1]
```diff
@@ -0,0 +1,116 @@
+<template>
+  <dx-toolbar>
+    <dx-item location="before">
+      <span class="toolbar-header">
+        <slot />
+      </span>
+    </dx-item>
+
+    <dx-item
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/analytics/components/analytics-toolbar.vue:9]: Looks like we can use slot here to send dx-item as children

File: packages/vue/src/pages/analytics/analytics-dashboard.vue [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+  <div class="view-wrapper">
+    <analytics-toolbar
+      ref="tabs"
+      :show-tabs="true"
+      @tab-change="tabChange($event)"
+    >
+      Dashboard
+    </analytics-toolbar>
+    <div class="tiles">
+      <card-analytics
```
[the remark this answers is not in the document]
Comment (dxvladislavvolkov) [reply packages/vue/src/pages/analytics/analytics-dashboard.vue:11]: The same problem with others analytics pages

File: packages/angular/src/app/pages/analytics-sales-report/analytics-sales-report.component.html [hunk 3]
```diff
@@ -82,14 +80,14 @@
       </dx-chart>
     </card-analytics>
 
-    <card-analytics titleText="Sales Perfomance" className="sales" [isLoading]="!salesByDateAndCategory"  [component]="chart">
+    <card-analytics titleText="Sales Perfomance" className="sales" [component]="chart">
```
Comment (dxvladislavvolkov) [inline packages/angular/src/app/pages/analytics-sales-report/analytics-sales-report.component.html:83]: Fix typo Perfomance -> Performance

File: packages/vue/src/pages/analytics/analytics-dashboard.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/pages/analytics/analytics-dashboard.vue]: fix: refactor

File: packages/angular/src/app/pages/analytics-sales-report/analytics-sales-report.component.html (hunk 3 above)
Comment (GoodDayForSurf) [reply packages/angular/src/app/pages/analytics-sales-report/analytics-sales-report.component.html:83]: fix

File: packages/angular/src/app/pages/analytics-geography/analytics-geography.component.html [hunk 4]
```diff
@@ -45,20 +44,19 @@
     <card-analytics
       titleText="Revenue Analysis by States"
       contentClass="sales-by-state-and-city"
-      [isLoading]="!salesByStateAndCity"
     >
-      <dx-data-grid [dataSource]="salesByStateAndCity" [height]="270">
-        <dxi-column 
-          caption="State" 
+      <dx-data-grid [dataSource]="salesByStateAndCity" [height]="270" [loadPanel]="{enabled: false}">
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/angular/src/app/pages/analytics-geography/analytics-geography.component.html:48]: In vue too

File: packages/angular/src/app/pages/analytics-sales-report/analytics-sales-report.component.ts [hunk 5]
```diff
@@ -73,46 +66,41 @@ export class AnalyticsSalesReportComponent implements OnInit, OnDestroy {
...
-        this.salesByDateAndCategory = salesByDateAndCategory;
-        this.salesByCategory = salesByCategory;
-      })
-      ),
-    ];
+    [
+      ['sales', this.service.getSales(startDate, endDate)],
+      ['salesByDateAndCategory', this.service.getSalesByOrderDate(groupBy)],
+    ].forEach(([dataName, loader]: [string, Observable<any>]) => {
+        this[dataName]  = null;
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/angular/src/app/pages/analytics-sales-report/analytics-sales-report.component.ts]: fix

File: packages/vue/src/pages/analytics/analytics-dashboard/analytics-dashboard.vue [hunk 6]
```diff
@@ -0,0 +1,133 @@
+<template>
+  <div class="view-wrapper">
+    <analytics-toolbar
+      ref="tabs"
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/analytics/analytics-dashboard/analytics-dashboard.vue:4]: Remove ref

**Review by groshenkovamarina — DISMISSED** [review]

File: packages/vue/src/pages/analytics/analytics-geography/analytics-geography.vue [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+const createMapCoords = (coords: string) => coords.split(', ').map((coord) => parseFloat(coord));
+
+const loadData = async (startDate: string, endDate: string) => {
+  salesByStateAndCity.value = null;
+  salesByState.value = null;
+  salesByStateMarkers.value = null;
+
+  salesByStateAndCity.value = await getSalesByStateAndCity(startDate, endDate);
+  salesByState.value = calcSalesByState(salesByStateAndCity.value);
+  salesByStateMarkers.value = {
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/analytics/analytics-geography/analytics-geography.vue:46]: Move the object to some function. Somethink like getSalesByStateMarkers

Comment (GoodDayForSurf) [reply packages/vue/src/pages/analytics/analytics-geography/analytics-geography.vue]: ok

**Review by groshenkovamarina — APPROVED** [review]

File: packages/vue/src/pages/analytics/analytics-dashboard/components/analytic-tile.vue [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+import { formatPrice } from '@/utils/formatters';
+import { Sales, SalesOrOpportunitiesByCategory } from '@/types/analytics';
+import { computed, withDefaults } from 'vue';
+import CardAnalytics from '../../components/card-analytics.vue';
+
+const props = withDefaults(
+  defineProps<{
+    data?: SalesOrOpportunitiesByCategory | Sales | null,
+    total?: string | null,
+    up: boolean,
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/analytics/analytics-dashboard/components/analytic-tile.vue:27]: Сan give a more descriptive name for the option?

**Review by dxvladislavvolkov — APPROVED** [review]

File: packages/vue/src/pages/analytics/analytics-dashboard/components/analytic-tile.vue (hunk 8 above)
Comment (GoodDayForSurf) [reply packages/vue/src/pages/analytics/analytics-dashboard/components/analytic-tile.vue:27]: yep.. in next approve

**Pull request merged**